core/storage/email/gae_models.py

```diff
@@ -212,3 +212,88 @@ def create(
             sender_email=sender_email, intent=intent, subject=subject,
             html_body=html_body, sent_datetime=sent_datetime)
         email_model_instance.put()
+
+
+class FeedbackEmailReplyToIDModel(base_models.BaseModel):
+    """This model stores unique_id for each <user, exploration, thread>
+    combination.
+
+    This unique_id is used in reply-to email address in outgoing feedback and
+    suggestion emails. The id/key of instances of this model has form of
+    [USER_ID].[EXPLORATION_ID].[THREAD_ID]
+    """
+    # reply_to ID that is used in repy-to email address.
+    reply_to_id = ndb.StringProperty(indexed=True, required=True)
+
+    @classmethod
+    def _generate_id(cls, user_id, exploration_id, thread_id):
+        return '.'.join([user_id, exploration_id, thread_id])
+
+    @classmethod
+    def _generate_unique_reply_to_id(cls):
+        for _ in range(base_models.MAX_RETRIES):
+            new_id = utils.convert_to_hash(
+                '%s' % (utils.get_random_int(base_models.RAND_RANGE)),
+                base_models.ID_LENGTH)
```

Comment:
Can we actually use a longer ID here? It's never actually read so I think longer would be better; maybe see what Google Docs or GitHub does.